src/check.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for running checks."""
+
+import logging
+from collections.abc import Iterable, Iterator
+from typing import NamedTuple, TypeGuard
+
+from .content import conflicts as content_conflicts
+from .content import diff as content_diff
+from .types_ import AnyAction, UpdateAction
+
+
+class Problem(NamedTuple):
+    """Details about a failed check.
+
+    Attrs:
+        path: Unique identifier for the file and discourse topic with the problem
+        description: A summary of what the problem is and how to resolve it.
+
+    """
+
+    path: str
+    description: str
+
+
+def _is_update_action(action: AnyAction) -> TypeGuard[UpdateAction]:
+    """Check whether an action is an UpdateAction.
+
+    Args:
+        action: The action to check.
+
+    Returns:
+        Whether the action is an UpdateAction.
+    """
+    return isinstance(action, UpdateAction)
+
+
+def _update_action_problem(action: UpdateAction) -> Problem | None:
```

Comment:
**suggestion** can we centralize the "conflicts/check" logic into the ContentChange.conflicts() method (related to another comment here) only? Now it seems that the check is somewhat split between this function and the `content_conflicts`, although really, all the logic also here only considers content_change attribute (and sub-attributes)

For better readability I'd just keep everything in a single place

Reply:
The purpose of this method is more than just the conflicts, although that is the use case so far. At a high level, it made sense to introduce this `check` module which takes actions and reports any problems with them. I can see a bunch of other use cases for this, e.g., spellcheck, that links work and so on

Reply:
yes, I see that. I would just keep logic in this as clean and readable as possible, and clearly factor out the different checks based on underlying data/class, e.g. 

```
def _update_action_problem(action: UpdateAction) -> Problem | None:
    action_errors = content_conflicts(action.content_change) or other_check(action.path) or other_check_2(...)
    
    if action_errors is None:
        return None
    
    return Problem(
            path=action.path,
            description=(
                "There are some problems: \n{action_errors}"
            ),
```

Anyhow, as before, this is a bit of matter of personal taste (I believe it increases readability, but other may feel differently), so definitely non-blocking. Resolving discussion